File: 2000 98 & ME/QuickStart/howto/samples/Xml/LoadDataSetXMLData/VB/LoadDataSetXMLData.vb
' Copyright (c) 2000, Microsoft Corporation. All Rights Reserved.

Option strict off

Imports Microsoft.VisualBasic
Imports System
Imports System.IO
Imports System.Data
Imports System.Xml

namespace HowTo.Samples.XML

public class LoadDataSetXMLDataSample

    private const m_Document as string = "books.xml"
    private const m_Schema as string = "books.xsd"
    private m_datadoc as XmlDataDocument 

    public sub New()        
    MyBase.new
        try
            Console.WriteLine("Creating an XmlDataDocument ...")
            m_datadoc = new XmlDataDocument()
            ParseSchema(m_Schema)
            DisplayTableStructure()

            m_datadoc.Load(m_Document)

            DisplayTables(m_datadoc.DataSet)
        
        catch e as exception
            Console.WriteLine ("Exception: " & e.ToString())
        end try
    end sub

    shared sub Main()
        Dim xmltransform = new LoadDataSetXMLDataSample()
    end sub

    ' Loads a specified schema into the DataSet
    public sub ParseSchema(schema as string)

        dim streamreader_schema as StreamReader = nothing
        try 
            streamreader_schema = new StreamReader(schema)
            Console.WriteLine("Reading Schema file ...")
            m_datadoc.DataSet.ReadXmlSchema(streamreader_schema)

        catch e as exception
            Console.WriteLine ("Exception: " & e.ToString())

        finally
            if not (streamreader_schema = nothing)
                streamreader_schema.Close()
            end if

        end try
    end sub

    ' Displays the DataSet tables structure
    private sub DisplayTableStructure()
    
        Console.WriteLine("Table structure ")
        Console.WriteLine()
        Console.WriteLine("Tables count=" & m_datadoc.DataSet.Tables.Count.ToString())
        dim i as Integer
        for i = 0 to (m_datadoc.DataSet.Tables.Count - 1)
        
            Console.WriteLine()
            Console.WriteLine("TableName='" & m_datadoc.DataSet.Tables(i).TableName + "'.")
            Console.WriteLine("Columns count=" & m_datadoc.DataSet.Tables(i).Columns.Count.ToString())

            Dim j As Integer
            for j = 0 to (m_datadoc.DataSet.Tables(i).Columns.Count - 1)            
                Console.WriteLine(Strings.chr(9) & "ColumnName='" & m_datadoc.DataSet.Tables(i).Columns(j).ColumnName.ToString() & "', type = " & m_datadoc.DataSet.Tables(i).Columns(j).DataType.ToString())
            Next
        Next
        
    end sub

    ' Displays the contents of the DataSet tables
    private sub DisplayTables(ds as DataSet)

        ' Navigate Dataset
        Console.WriteLine()
        Console.WriteLine("Content of Tables ...")

        dim table as object
        for each table in ds.Tables
        
            Console.WriteLine("TableName = " & table.TableName.ToString())
            Console.WriteLine ("---------")
            Console.WriteLine("Columns ...")

            dim column as object
            for each column in table.Columns
            
                Console.Write("{0,-22}",column.ColumnName.ToString())
            next
            Console.WriteLine()
            Console.WriteLine("Number of rows = {0}", table.Rows.Count.ToString())
            Console.WriteLine("Rows ...")

            dim row as object 
            for each row in table.Rows

            dim value as object            
                for each value in row.ItemArray
                    Console.Write("{0,-22}",value.ToString())
                Next
                Console.WriteLine()
            Next
            Console.WriteLine()
        Next
    end sub
end class 
End namespace 

File: 2000 98 & ME/QuickStart/howto/samples/Xml/LoadDataSetXMLData/LoadDataSetXMLData.cs
/* Copyright (c) 2000, Microsoft Corporation. All Rights Reserved.*/

namespace HowTo.Samples.XML
{

using System;
using System.IO;
using System.Data;
using System.Xml;

public class LoadDataSetXMLDataSample
{
    private const String m_Document = "books.xml";
    private const String m_Schema = "books.xsd";
    private XmlDataDocument m_datadoc;

    public LoadDataSetXMLDataSample()
    {        
        try
        {
            Console.WriteLine("Creating an XmlDataDocument ...");
            m_datadoc = new XmlDataDocument();
            ParseSchema(m_Schema);
            DisplayTableStructure();

            m_datadoc.Load(m_Document);

            DisplayTables(m_datadoc.DataSet);
        }
        catch (Exception e)
        {
            Console.WriteLine ("Exception: {0}", e.ToString());
        }
    }

    public static void Main()
    {
        LoadDataSetXMLDataSample xmltransform = new LoadDataSetXMLDataSample();
    }

    // Loads a specified schema into the DataSet
    public void ParseSchema(String schema)
    {
        StreamReader streamreader_schema = null;

        try 
        {
            Console.WriteLine("Reading Schema file ...");
            streamreader_schema = new StreamReader(schema);
            m_datadoc.DataSet.ReadXmlSchema(streamreader_schema);
        }

        catch (Exception e)
        {
            Console.WriteLine ("Exception: {0}", e.ToString());
        }

        finally 
        {
            if (streamreader_schema != null)
                streamreader_schema.Close();
        }
    }    

    // Displays the DataSet tables structure
    private void DisplayTableStructure()
    {
        Console.WriteLine("\r\nTable structure \r\n");
        Console.WriteLine("Tables count=" + m_datadoc.DataSet.Tables.Count.ToString());
        for (int i = 0; i < m_datadoc.DataSet.Tables.Count; i++)
        {
            Console.WriteLine("\tTableName='" + m_datadoc.DataSet.Tables[i].TableName + "'.");
            Console.WriteLine("\tColumns count=" + m_datadoc.DataSet.Tables[i].Columns.Count.ToString());

            for (int j = 0; j < m_datadoc.DataSet.Tables[i].Columns.Count; j++)
            {
                Console.WriteLine("\t\tColumnName='" + m_datadoc.DataSet.Tables[i].Columns[j].ColumnName + "', type = " + m_datadoc.DataSet.Tables[i].Columns[j].DataType.ToString());
            }
        }
    }

    // Displays the contents of the DataSet tables
    private void DisplayTables(DataSet dataset)
    {
        // Navigate Dataset
        Console.WriteLine("Content of Tables ...\r\n");

        foreach(DataTable table in dataset.Tables)
        {
            Console.WriteLine("TableName = " + table.TableName);
            Console.WriteLine ("{0}", "---------");
            Console.WriteLine("Columns ...\r\n");

            foreach(DataColumn column in table.Columns)
            {
                Console.Write("{0,-22}",column.ColumnName);
            }
            Console.WriteLine();
            Console.WriteLine("\r\nNumber of rows = {0}", table.Rows.All.Length);
            Console.WriteLine("Rows ...\r\n");

            foreach(DataRow row in table.Rows)
            {
                foreach(Object value in row.ItemArray)
                {
                    Console.Write("{0,-22}",value.ToString());
                }
                Console.WriteLine();
            }
            Console.WriteLine();
        }
    }
} // End class LoadDataSetXMLDataSample
} // End namespace HowTo.Samples.XML
